pkg/construct2/graph_io.go

```diff
@@ -46,7 +46,12 @@ func (g YamlGraph) MarshalYAML() (interface{}, error) {
 			errs = errors.Join(errs, err)
 			continue
 		}
-
+		if r.Imported {
+			if r.Properties == nil {
+				r.Properties = make(Properties)
+			}
```

Comment:
Properties should never be `nil`, while it's not documented, it's part of the contract.

Reply:
i ran into a nil pointer exception during my testing which is why i added this. I didnt look into why the properties were nil too deeply though

Reply:
Sounds like a bug somewhere. I'd rather fix the cause of breaking the contract than start sprinkling guards against breaking the contract. (we should also document that `Resource.Properties` is never nil on the godoc)